src/components/ReviewConfirm/IncorporationDateTime.vue

```diff
@@ -0,0 +1,368 @@
+            </v-col>
+            <v-col cols="12" sm="6" md="3">
+              <v-select
+                id="am-pm-selector"
+                :items="timePeriod"
+                v-model="selectPeriod"
+                :disabled="!isFutureEffective"
+                filled
+              ></v-select>
+            </v-col>
+            <v-col cols="12" sm="6" md="2" class="label-col">
+              <span class="time-zone-label" :class="{ 'disabled': !isFutureEffective }">Pacific Time</span>
+            </v-col>
+          </v-row>
+        </div>
+      </v-col>
+    </v-row>
+  </v-card>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Mixins, Watch, Emit, Prop } from 'vue-property-decorator'
+import { Action } from 'vuex-class'
+
+// Mixins
+import { DateMixin } from '@/mixins'
+
+// Constants
+import { ISIMMEDIATE, ISFUTUREEFFECTIVE } from '@/constants'
+
+// Interfaces
+import { ActionBindingIF, DateTimeIF } from '@/interfaces'
+
+@Component({})
+export default class IncorporationDateTime extends Mixins(DateMixin) {
+  @Prop()
+  private incorporationDateTime!: DateTimeIF
+
+  // Global Actions
+  @Action setIsImmediate!: ActionBindingIF
+  @Action setIsFutureEffective!: ActionBindingIF
+
+  // Local properties
+  private isImmediate: boolean = false
+  private isFutureEffective: boolean = false
+
+  // Date properties
+  private selectDate: string = ''
+  private dateText: string = ''
+  private datePicker: string = ''
+
+  // Time properties
+  private selectHour: string = ''
+  private selectMinute: string = ''
+  private selectPeriod: string = 'AM'
+
+  // Date Time Selectors
+  private hours: Array<string> = [...Array(12).keys()]
+    .map(num => (++num).toLocaleString('en-US'))
+  private minutes: Array<string> = [...Array(60).keys()]
+    .map(num => num.toLocaleString('en-US', { minimumIntegerDigits: 2 }))
+  private timePeriod: Array<string> = ['AM', 'PM']
+
+  /** Lifecycle Hook to run when component mounts */
+  mounted (): void {
+    this.deconstructDateTime()
+  }
+
+  /**  Clear the selected Date and time */
+  private clearDateTimes (): void {
+    // Clear Date
+    this.dateText = ''
+    this.datePicker = ''
+
+    // Clear Times
+    this.selectHour = ''
+    this.selectMinute = ''
+    this.selectPeriod = 'AM'
+
+    // Clear Future Effective from store / fee summary
+    this.setIsFutureEffective(false)
+    this.emitDateTime('')
+  }
+
+  /** Construct the Date Object for storage */
+  private constructDate (): void {
+    if (this.isFutureEffective) {
+      // Format the selected date string and create Date
+      const dateString = this.dateText.replace(/-/g, ', ')
+      const dateToValidate = new Date(dateString)
+
+      // Create references & Apply time period
+      let hours = +this.selectHour
+      const minutes = +this.selectMinute
+
+      if (this.selectPeriod === 'AM' && +this.selectHour === 12) {
+        dateToValidate.setDate(dateToValidate.getDate() - 1)
```

Comment:
What is the purpose of this?

Reply:
Since we set each Date, Hours, minutes and AmPm separately to construct a DateTime, it was tricky setting to 12:00AM on the selected Date. What it was doing when i chose to set the Hours & Minutes of a selected day wasn't taking into consideration the AM/PM. So when i user selected 12 hours AND AM... that essentially equates to 00:00 time. 
So to get that time on the selected day, i had to set it to the day prior and Add the 12hrs to reset the clock essentially.  This is mainly a result of the hour selector only being 12hr selector opposed to a 24hr and using the 'AmPm' field.
It was tricky having them all in isolation but working together :P. 
Does this make sense? Could this be done better?

Reply:
I think I  got it. Is it converting to previous day 24:00? Is there any reason for not converting to current day 00:00? Just curious

Reply:
Great question, it IS converting to prior day at 24:00. You're right for sure, i could have set it explicitly, but i think i just structured like this so i only have to `Date.setHours()` once inside the function. 
i was using the `hours` selected to set the hour and there is no `0` in the selectors, so essentially selecting `12`th hour  in conjunction with `am` equates to 00 on the same day selected ( or in this case the code is 24:00 the previous day ). 

phew.. long winded, sorry 😆